### PR TITLE
Impl guest side verification

### PR DESCRIPTION
**Pull request by ercecan**
# Description
**MERGE AFTER** #1336 
Initial pr that prepares the foundation for #1306 
After merged, for testing Risc0's `FakeReceipt`s will be used with `RISC0_DEV_MODE` enabled

## Linked Issues
- Related to #1306

### CODE REVIEW

File: crates/sovereign-sdk/adapters/mock-zkvm/src/lib.rs [hunk 1]
```diff
@@ -171,14 +171,14 @@ impl<ValidityCond: ValidityCondition> sov_rollup_interface::zk::ZkvmHost
...
-        Ok(sov_rollup_interface::zk::Proof::PublicInput(data))
+        Ok(sov_rollup_interface::zk::Proof::Mock(data))
     }
 
     fn extract_output<Da: sov_rollup_interface::da::DaSpec, Root: BorshDeserialize>(
         proof: &Proof,
     ) -> Result<sov_rollup_interface::zk::StateTransition<Da, Root>, Self::Error> {
         match proof {
-            sov_rollup_interface::zk::Proof::PublicInput(pub_input) => {
+            sov_rollup_interface::zk::Proof::Mock(pub_input) => {
```
Comment (eyusufatik) [inline crates/sovereign-sdk/adapters/mock-zkvm/src/lib.rs]: call this fake receipt or test receipt

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/sovereign-sdk/rollup-interface/src/state_machine/zk/mod.rs [hunk 2]
```diff
@@ -26,8 +26,9 @@ use crate::spec::SpecId;
 /// The ZK proof generated by the [`ZkvmHost::run`] method.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
 pub enum Proof {
+    /// Mock proof
     /// Only public input was generated.
```
Comment (eyusufatik) [inline crates/sovereign-sdk/rollup-interface/src/state_machine/zk/mod.rs]: remove all of this public input stuff, now we have the receipt for execute mode so the output has to be same

File: crates/prover-services/src/parallel/mod.rs [hunk 3]
```diff
@@ -156,7 +156,7 @@ where
         // TODO: maybe extract this to Vm?
         // Extract journal
         let journal = match proof {
-            Proof::PublicInput(journal) => {
+            Proof::FakeReceipt(journal) => {
                 let journal: Journal = bincode::deserialize(&journal)?;
```
Comment (eyusufatik) [inline crates/prover-services/src/parallel/mod.rs]: this should be FakeReceipt now?

**Review by eyusufatik — CHANGES_REQUESTED** [review]

**Comment by ercecan** [comment]
Closing in favor of #1402

**Pull request closed without merging**